src/fileseq/frameset.py

```diff
@@ -113,14 +116,16 @@ def __init__(self, frange):
             # in all other cases, cast to a string
             else:
                 try:
-                    frange = str(frange)
+                    frange = asString(frange)
                 except Exception as err:
                     msg = 'Could not parse "{0}": cast to string raised: {1}'
                     raise ParseException(msg.format(frange, err))
 
         # we're willing to trim padding characters from consideration
         # this translation is orders of magnitude faster than prior method
-        self._frange = str(frange).translate(None, ''.join(PAD_MAP.keys()))
+        self._frange = u''.join(
+            [char for char in asString(frange) if not char in PAD_MAP]
+        )
```

Comment:
Looks like there is an equivalent approach in Py3:  

```
asString(frange).translate(str.maketrans('', '', ''.join(PAD_MAP)))
```

Is it trivial to do a py2/3 branch via six at this point, instead of dropping the logic down to a loop over each character?

Reply:
Then the PY2 implementation is a problem, because now that we're dealing with unicode strings, there's no useful combination of translate/maketrans that we can use. We can cast frange to a str and then back to a unicode, but that's not very satisfying.

Check this out though:
https://gist.github.com/donalm/5dd45169ef7d0e7b6041613bff92be61